mongo/options/changestreamoptions.go

```diff
@@ -7,76 +7,95 @@
 package options
 
 import (
-	"go.mongodb.org/mongo-driver/bson/primitive"
 	"time"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
 )
 
-// ChangeStreamOptions represents all possible options to a change stream
+// ChangeStreamOptions represents options that can be used to configure a Watch operation.
 type ChangeStreamOptions struct {
-	BatchSize            *int32               // The number of documents to return per batch
-	Collation            *Collation           // Specifies a collation
-	FullDocument         *FullDocument        // When set to ‘updateLookup’, the change notification for partial updates will include both a delta describing the changes to the document, as well as a copy of the entire document that was changed from some time after the change occurred.
-	MaxAwaitTime         *time.Duration       // The maximum amount of time for the server to wait on new documents to satisfy a change stream query
-	ResumeAfter          interface{}          // Specifies the logical starting point for the new change stream
-	StartAtOperationTime *primitive.Timestamp // Ensures that a change stream will only provide changes that occurred after a timestamp.
-	StartAfter           interface{}          // Specifies a resume token. The started change stream will return the first notification after the token.
+	// The maximum number of documents to be included in each batch returned by the server.
+	BatchSize *int32
+
+	// Specifies a collation to use for string comparisons during the operation. This option is only valid for MongoDB
+	// versions >= 3.4. For previous server versions, the driver will return an error if this option is used. The
+	// default value is nil, which means the default collation of the collection will be used.
+	Collation *Collation
+
+	// Specifies whether the updated document should be returned in change notifications for update operations along
+	// with the deltas describing the changes made to the document. The default is options.Default, which means that
+	// the updated document will not be included in the change notification.
+	FullDocument *FullDocument
+
+	// The maximum amount of time that the server should wait for new documents to satisfy a tailalbe cursor query.
+	MaxAwaitTime *time.Duration
+
+	// A document specifying the logical starting point for the change stream. Only change starting with a log entry
+	// immediately after the resume token will be returned.
+	ResumeAfter interface{}
+
+	// If specified, the change stream will only return changes that occurred at or after the given timestamp. This
+	// option is only valid for MongoDB versions >= 4.0.
+	StartAtOperationTime *primitive.Timestamp
```

Comment:
Is this exclusive to ResumeAfter/StartAfter?  If so, indicate that.

Reply:
Not sure what you mean. `StartAtOperationTime` is a completely separate option than `ResumeAfter`/`StartAfter`. It takes a timestamp, whereas the other two take a BSON document.

Reply:
Right, but I thought that all three of these options were mutually exclusive.  If you specify one, the others must be nil.  Is that not true?

Reply:
Good catch. I added a note to each of the three saying that if it is set, the other two must not be.